### DISCUSSION TITLE
Resetting user password using the database

### DISCUSSION THREAD

**Discussion by bhavin192** [plausible/analytics#819] [Self-Hosted Support] [answered]
The reset password method was not working for an account as the email provider was rejecting the emails. I updated the email address by changing it in the database, so that the password reset link can be send (other email provider had no issues). Is there any other way to reset the password by making changes to the database or using some other method than email?

**Comment by Vigasaurus** [comment, marked as answer]
https://github.com/plausible/analytics/discussions/736 you'd need to combine a bit of the steps here, but the building blocks are there. You'd need to use the `get_by()` and then the set_password()`

**Comment by binary-person** [comment]
I stumbled upon this issue since I couldn't get the email server to work, and thus, couldn't get the reset password to work. Posting this here for future me and other people:

1. find your main plausible instance name by doing `docker ps`
2. run `docker exec -it plausible_plausible_1 /app/bin/plausible remote` (for me, the name was `plausible_plausible_1`)
3. run `Plausible.Repo.get_by(Plausible.Auth.User, email: "youremail@domain.com") |> Plausible.Auth.User.set_password("yourpassword") |> Plausible.Repo.update`

**Reply by HassanAthmani** [reply]
This worked perfectly.

**Reply by Xevion** [reply]
Not a fan of this because on Railway, there's no way to execute arbitrary commands to a given Docker instance. Instead, I ended up just getting the SMTP server to work; it's not hard to do it with FastMail anyways.
That said, if you don't know your email or if it's a bad one - you're kinda screwed.

**Reply by akoskm** [reply]
> I stumbled upon this issue since I couldn't get the email server to work, and thus, couldn't get the reset password to work. Posting this here for future me and other people:
> 
>     1. find your main plausible instance name by doing `docker ps`
> 
>     2. run `docker exec -it plausible_plausible_1 /app/bin/plausible remote` (for me, the name was `plausible_plausible_1`)
> 
>     3. run `Plausible.Repo.get_by(Plausible.Auth.User, email: "youremail@domain.com") |> Plausible.Auth.User.set_password("yourpassword") |> Plausible.Repo.update`

Works even after a year! 🙌 Thanks

**Comment by chrissound** [comment]
If it helps anyone else, in case you've forgotten your email as well: you can run `Plausible.Repo.all(Plausible.Auth.User)`.

**Comment by StevKast** [comment]
This just helped me out, thanks!
